book/website/collaborating_github/reviewing-contributions.md

```diff
@@ -0,0 +1,89 @@
+Knowing the working of Version control systems, [markdown](https://guides.github.com/features/mastering-markdown/) and open source [workflow](https://guides.github.com/introduction/flow/) helps to a great extent in carrying out all these tasks.
+
+## Ownership and Permissions 
+In any team or organization owned project, there are different contributors. 
+
+- Some start the project and are generally, the owners of the project. The Codeowners who have the rights over who can make reviews and suggest changes to contributions. The individuals or teams who are codeowners for a certain project are specified at the time of project creation in a [file](https://help.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners) of the project.  
+They can make other contributors as maintainers, reviewers or members working on the project. They can also involve other codeowners and are responsible for granting permissions. 
+
+- Maintainers are involved with creating issues and pull requests whenever required. They keep the codebase updated and help in reviewing contributions. They can merge pull requests.  
+They can also request reviews from someone else. There are various [levels of permissions](https://help.github.com/en/github/setting-up-and-managing-organizations-and-teams/repository-permission-levels-for-an-organization) for a project in GitHub.
+
+## Reviewing process
+- Any code base or set of changes must be reviewed before merging. If the project os co-owned by many people, the review process is done by handling it in the form of a [code review assignment](https://help.github.com/en/github/setting-up-and-managing-organizations-and-teams/managing-code-review-assignment-for-your-team) in which certain members of the team are chosen for the review.  
+This member subset of team is selected with the help of algorithms after reviewing is enabled and review assignment is configured for a particular project.  
+- The review process is dependent on the testing of a particular pull request by a maintainer or admin. The admins can easily [test](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/checking-out-pull-requests-locally) the changes made in their local systems and suggest changes required.  
+- After this any set of changes through pull requests can be reviewed. When the review process is completed the reviews are [approved](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/approving-a-pull-request-with-required-reviews) or [dismissed](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/dismissing-a-pull-request-review) according to the outcome. 
+
+## Merging changes
+Pull requests created by contributors can be easily merged by maintainers after the review. There are different ways of merging a PR:
+- Merge pull request- Merges all commits to base branch. This keeps all commits made separated. 
```

Comment:
```suggestion
- Merge pull request- Merges all commits to base branch. This keeps all commits made separated. 
```
This method merges all commits as a single merge commit. https://rietta.com/blog/github-merge-types/#:~:text=Create%20a%20Merge%20Commit,commits%20are%20One%20and%20Two%20.

Reply:
Yes the merge commit is a single one in both "Merge pull request" and "Squash and Merge" methods but if there are multiple commits made which are referenced to be merged through a PR the first option will keep them as separate and the second one would squash them into one. Should I make the description a little longer?

Reply:
Yes I think so. As it currently reads, why merge and squash commits differ is not obvious.

Reply:
I have made changes in the text. Can you please check and update accordingly? :slightly_smiling_face: